WebService.Test/v1/Controllers/StatusControllerTest.cs

```
@@ -0,0 +1,99 @@
+        private readonly Mock<IStorageAdapterClient> storage;
+        private readonly Mock<ISimulations> simulations;
+        private readonly Mock<ILogger> logger;
+        private readonly Mock<IServicesConfig> servicesConfig;
+        private readonly Mock<IDeploymentConfig> deploymentConfig;
+        private readonly Mock<IIotHubConnectionStringManager> connectionStringManager;
+        private readonly Mock<ISimulationRunner> simulationRunner;
+        private readonly StatusController target;
+
+        public StatusControllerTest(ITestOutputHelper log)
+        {
+            this.preprovisionedIotHub = new Mock<IPreprovisionedIotHub>();
+            this.storage = new Mock<IStorageAdapterClient>();
+            this.simulations = new Mock<ISimulations>();
+            this.logger = new Mock<ILogger>();
+            this.servicesConfig = new Mock<IServicesConfig>();
+            this.deploymentConfig = new Mock<IDeploymentConfig>();
+            this.connectionStringManager = new Mock<IIotHubConnectionStringManager>();
+            this.simulationRunner = new Mock<ISimulationRunner>();
+
+            this.target = new StatusController(
+                this.preprovisionedIotHub.Object,
+                this.storage.Object,
+                this.simulations.Object,
+                this.logger.Object,
+                this.servicesConfig.Object,
+                this.deploymentConfig.Object,
+                this.connectionStringManager.Object,
+                this.simulationRunner.Object);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public async Task GetTest()
+        {
+            // Arrange
+            SetupSimulationForRunner();
+
+            this.simulationRunner
+                .Setup(x => x.GetActiveDevicesCount())
+                .Returns(5);
+
+            // Act
+            var result = await this.target.Get();
+
+            // Assert
+            Assert.Equal(3, result.Properties.Count);
```

which properties are you validating ? Can you add more checks ? And tests for other cases if different properties are expected ?

---

Currently is validating `ActiveDeviceCount`. Will add more checks when add `telemetry metrics`.

---

testing for the count will mean that this test will fail everytime we add a new property. The test should fail only if the "device count" value is missing